packages/zoltan/src/hier/hier.c

```diff
@@ -1552,6 +1552,7 @@ int Zoltan_Hier(
       printf("HIER: Proc %d computing part %d of %d at level %d\n",
              zz->Proc, hpp.part_to_compute, hpp.num_parts, hpp.level);
     }
+    MPI_Barrier(zz->Communicator);
```

Comment:
Is the barrier needed here, or is it here for output purposes?

Reply:
Yes, I believe it was for output that was deleted. Removed.